src/main/java/org/testcontainers/containers/TarantoolContainerClientHelper.java

```diff
@@ -1,88 +1,141 @@
+        container.copyFileToContainer(MountableFile.forClasspathResource(scriptResourcePath), containerPath);
+        return executeCommand(String.format("return dofile('%s')", containerPath), sslContext);
     }
 
-    private TarantoolClient<TarantoolTuple, TarantoolResult<TarantoolTuple>> createClient() {
-        return clientBuilder
-                .withCredentials(container.getUsername(), container.getPassword())
-                .withAddress(container.getHost(), container.getPort())
-                .build();
-    }
+    public <T> T executeScriptDecoded(String scriptResourcePath, SslContext sslContext) throws IOException, InterruptedException, ExecutionException {
+        Container.ExecResult result = executeScript(scriptResourcePath, sslContext);
 
-    /**
-     * Configure or return an already configured client connected to a Cartridge router
-     *
-     * @return a configured client
-     */
-    public TarantoolClient<TarantoolTuple, TarantoolResult<TarantoolTuple>> getClient() {
-        if (!container.isRunning()) {
-            throw new IllegalStateException("Cannot connect to Tarantool instance in a stopped container");
-        }
-        if (clientHolder.get() == null) {
-            clientHolder.compareAndSet(null, createClient());
+        if (result.getExitCode() != 0) {
+
+            if (result.getExitCode() == 3 || result.getExitCode() == 1) {
+                throw new ExecutionException(String.format(EXECUTE_SCRIPT_ERROR_TEMPLATE,
+                    scriptResourcePath, result.getExitCode(),
+                    result.getStderr(), result.getStdout()),
+                    new Throwable());
+            }
+
+            throw new IllegalStateException(String.format(EXECUTE_SCRIPT_ERROR_TEMPLATE,
+                scriptResourcePath, result.getExitCode(),
+                result.getStderr(), result.getStdout()));
         }
-        return clientHolder.get();
+
+        return yaml.load(result.getStdout());
     }
 
-    public CompletableFuture<List<?>> executeScript(String scriptResourcePath) {
+    public Container.ExecResult executeCommand(String command, SslContext sslContext) throws IOException, InterruptedException {
         if (!container.isRunning()) {
-            throw new IllegalStateException("Cannot execute scripts in stopped container");
+            throw new IllegalStateException("Cannot execute commands in stopped container");
         }
 
-        String scriptName = Paths.get(scriptResourcePath).getFileName().toString();
-        String containerPath = normalizePath(Paths.get(TMP_DIR, scriptName));
-        container.copyFileToContainer(MountableFile.forClasspathResource(scriptResourcePath), containerPath);
-        return executeCommand(String.format("return dofile('%s')", containerPath));
+        command = command.replace("\"", "\\\"");
+        command = command.replace("\'", "\\\'");
+
+        String bashCommand;
+        // No SSL
+        if (sslContext == null) {
```

Comment:
I think this type of comment code style is more appropriate in this case
```suggestion
        if (sslContext == null) { // No SSL
```